azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/parser/ConfigParser.java

```diff
@@ -65,7 +68,22 @@ public ContainerAppDraft.ImageConfig getImageConfigFromContainers(ContainerAppCo
         }
         if (containers.get(0).getDeploymentType() == DeploymentType.CODE || containers.get(0).getDeploymentType() == DeploymentType.ARTIFACT) {
             ContainerAppDraft.BuildImageConfig buildImageConfig = new ContainerAppDraft.BuildImageConfig();
-            buildImageConfig.setSource(Paths.get(containers.get(0).getDirectory()));
+            Path source = null;
+            if (containers.get(0).getDirectory() == null) {
```

Comment:
We have multiple places using `containers.get(0)`, better to pull it to a variable.